components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/basicauth/BasicAuthCredentialValidator.java

```diff
@@ -367,27 +255,35 @@ public boolean validateScopes(String username, OpenAPI openAPI, MessageContext s
         } else if (APIConstants.GRAPHQL_API.equals(synCtx.getProperty(APIConstants.API_TYPE))) {
             HashMap<String, String> operationScopeMappingList = (HashMap<String, String>) synCtx
                     .getProperty(APIConstants.SCOPE_OPERATION_MAPPING);
-            HashMap<String, ArrayList<String>> scopeRoleMappingList = (HashMap<String, ArrayList<String>>) synCtx
-                    .getProperty(APIConstants.SCOPE_ROLE_MAPPING);
             String[] operationList = ((String) synCtx.getProperty(APIConstants.API_ELECTED_RESOURCE)).split(",");
             for (String operation : operationList) {
                 String operationScope = operationScopeMappingList.get(operation);
                 if (operationScope != null) {
-                    ArrayList<String> operationRoles = scopeRoleMappingList.get(operationScope);
-                    boolean userHasOperationRole = false;
-                    for (String role : userRoles) {
-                        for (String operationRole : operationRoles) {
-                            if (operationRole.equals(role)) {
-                                userHasOperationRole = true;
-                                break;
+                    if (scopeMap.containsKey(operationScope)) {
+                        List<String> operationRoles = scopeMap.get(operationScope).getRoles();
+                        boolean userHasOperationRole = false;
+                        if (operationRoles.isEmpty()) {
+                            userHasOperationRole = true;
+                        } else {
+                            for (String role : userRoleList) {
+                                for (String operationRole : operationRoles) {
```

Comment:
instead 2nd for lopp can't we check
```suggestion
                   if (operationRoles.contains(role)) {
                   }           
```